File: Peperino/Dtos/CheckList/CheckListOutDto.cs
using Peperino.Core.EntityFramework.Entities;
using Peperino.Dtos.UserGroup;
using System.ComponentModel.DataAnnotations;

namespace Peperino.Dtos.CheckList
{
    public class CheckListOutDto
    {
        [Required]
        public int Id { get; set; }

        public RoomOutDto? Room { get; set; }

        [Required]
        public string Name { get; set; } = string.Empty;

        [Required]
        public string Slug { get; set; } = string.Empty;

        [Required]
        public List<BaseCheckListItemOutDto> Entities { get; set; } = new List<BaseCheckListItemOutDto>();

        [Required]
        public AccessLevel AccessLevel { get; set; }

        [Required]
        public bool IsFavorite { get; set; }
    }
}

File: peperino.ui/src/lib/api/models/CheckListOutDto.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { AccessLevel } from './AccessLevel';
import type { BaseCheckListItemOutDto } from './BaseCheckListItemOutDto';
import type { RoomOutDto } from './RoomOutDto';

export type CheckListOutDto = {
    id: number;
    room: RoomOutDto;
    name: string;
    slug: string;
    entities: Array<BaseCheckListItemOutDto>;
    accessLevel: AccessLevel;
    isFavorite: boolean;
};
